pywemo/ouimeaux_device/api/rules.py

```diff
@@ -0,0 +1,695 @@
+"""Access and manipulate the on-device rules sqlite database."""
+
+import base64
+import contextlib
+import io
+import logging
+import sqlite3
+import tempfile
+import zipfile
+from types import MappingProxyType
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    FrozenSet,
+    List,
+    Mapping,
+    Optional,
+    Tuple,
+)
+
+import requests
+
+# RuleDevicesRow.*Action values.
+ACTION_TOGGLE = 2.0
+ACTION_ON = 1.0
+ACTION_OFF = 0.0
+
+LOG = logging.getLogger(__name__)
+
+# RulesRow.Type values.
+RULE_TYPE_LONG_PRESS = "Long Press"
+
+VIRTUAL_DEVICE_UDN = "uuid:Socket-1_0-PyWemoVirtualDevice"
+
+
+class DatabaseRow:
```

Comment:
This looks like generic ORM style code (although good code).

I haven't use the Python sqllite bindings -but isn't there a DRY way to do this?